File: Core_Java/Exceptions(CookieMonster)/src/main/java/models/CookieEatingMonster.java
package models;

import exceptions.NotACookieException;


//This Class will model a monster that only eats cookies... A cookie monster if you will
//It has methods that will take in a Food object, and throw an exception if that food is not a cookie (boolean isCookie == false)
public class CookieEatingMonster {


    public void eatCookieWithTryCatch(Food food){

        try{
            if(food.isCookie){
                System.out.println("Cookie Eating Monster Eats a Cookie!!");
            }else{
                throw new NotACookieException("I only eat COOKIES. Feed me a cookie.");
            }

        } catch(NotACookieException e){
            System.out.println("NotACookieException Caught");
            e.printStackTrace();
        }

    }

    public void eatCookieWithThrows(Food food) throws NotACookieException{

        if(food.isCookie){
            System.out.println("Thanks for the cookie :)");
        }else {
            throw new NotACookieException("Do not feed me anything other than cookies.");
        }

    }



}

File: Python/Exceptions/app/cookie_eating_monster.py
"""
This Class represents a monster that eats cookies... a Cookie Monster if you will

He likes when you feed him cookies :)
He DOES NOT like when you feed him non-cookies.
    He will RAISE EXCEPTIONS if you don't feed him a cookie
"""
# TODO: If you want to review OOP, you can do a Food class with an "is_cookie" boolean
# TODO: Or make them do it themselves for a small practice/challenge

class CookieEatingMonster:

    def eat_cookie(self, food):
        if food != "cookie":
            raise NotACookieError(f"Disgusting. {food}.")
        else:
            print("Yum! Thanks for the cookie!")


# Custom Exception - NotACookieError - thrown when fed a non-cookie
class NotACookieError(Exception):
    def __init__(self, message):
        self.message = message
